src/gentropy/dataset/variant_index.py

```diff
@@ -1,78 +1,327 @@
-"""Variant index dataset."""
+"""Dataset definition for variant annotation."""
+
 from __future__ import annotations
 
 from dataclasses import dataclass
 from typing import TYPE_CHECKING
 
-import pyspark.sql.functions as f
+from pyspark.sql import functions as f
 
 from gentropy.common.schemas import parse_spark_schema
-from gentropy.common.spark_helpers import nullify_empty_array
+from gentropy.common.spark_helpers import (
+    get_record_with_maximum_value,
+    normalise_column,
+)
 from gentropy.dataset.dataset import Dataset
-from gentropy.dataset.study_locus import StudyLocus
+from gentropy.dataset.gene_index import GeneIndex
+from gentropy.dataset.v2g import V2G
 
 if TYPE_CHECKING:
+    from pyspark.sql import Column, DataFrame
     from pyspark.sql.types import StructType
 
-    from gentropy.dataset.variant_annotation import VariantAnnotation
-
 
 @dataclass
 class VariantIndex(Dataset):
-    """Variant index dataset.
-
-    Variant index dataset is the result of intersecting the variant annotation dataset with the variants with V2D available information.
-    """
+    """Variant annotation object based on the parsed VEP output."""
 
     @classmethod
     def get_schema(cls: type[VariantIndex]) -> StructType:
-        """Provides the schema for the VariantIndex dataset.
+        """Provides the schema for the variant annotation dataset.
 
         Returns:
-            StructType: Schema for the VariantIndex dataset
+            StructType: Schema for the VariantAnnotationNew dataset
         """
         return parse_spark_schema("variant_index.json")
 
-    @classmethod
-    def from_variant_annotation(
-        cls: type[VariantIndex],
-        variant_annotation: VariantAnnotation,
-        study_locus: StudyLocus,
-    ) -> VariantIndex:
-        """Initialise VariantIndex from pre-existing variant annotation dataset.
+    def add_annotation(self: VariantIndex, annotation: DataFrame) -> VariantIndex:
+        """Add annotation to the dataset.
+
+        At this point the variant annotation can be extended with extra cross-references and allele frequencies.
+
+        Args:
+            annotation (DataFrame): Annotation to add to the dataset
+
+        Returns:
+            VariantIndex: VariantIndex dataset with the annotation added
+        """
+        # Assert that the provided annotation has unique variant identifiers:
+        assert (
+            annotation.select("variantId").distinct().count() == annotation.count()
+        ), "Annotation must have unique variant identifiers."
+
+        # Rename columns to avoid conflicts:s
+        renamed_columns = {
+            col: f"annotation_{col}" if col != "variantId" else col
+            for col in annotation.columns
```

Comment:
Would you mind explaining?

Reply:
Not applicable anymore. I was suggesting to have as an argument which cols you wanted to include for annotation. So that you don't need to loop over all cols, just the ones you specify. As i said, not applicable anymore.